gocd/templates/gocd-agent-controller.yaml

```diff
@@ -190,3 +203,8 @@ spec:
       hostAliases:
 {{ toYaml .Values.agent.hostAliases | indent 8 }}
     {{- end }}
+{{- if and (eq .Values.agent.controller.kind "StatefulSet") .Values.agent.persistence.claimTemplates }}
```

Comment:
Should you need `agent.persistence.enabled` to make this apply? It's a little weird because the templates are under `agent.persistence` so it kinda feels like it should respect this, even though it's redundant and I guess the existing value has a different meaning. 

Or is the idea to allow both the normal `/home/go` volume as well as the stateful set ones (largely intended for `/godata`) in the same setup?

I'm wondering whether it adds more confusion to have this "inside" the existing `persistence` setup, but can't really think of a better idea; other than improving the docs.

Reply:
It is confusing, I agree that there is no 100% clean way beside doing a bigger yaml refactoring.

The way I took was to make them behave like `agent.persistence.extraVolumeMounts`, which are similarly "persistence, but not the kind that's attached to `enabled`".